onnxruntime/core/framework/kernel_type_str_resolver.cc

```diff
@@ -0,0 +1,234 @@
+  for (const Node& node : graph.Nodes()) {
+    ORT_RETURN_IF_ERROR(RegisterNodeOpSchema(node));
+
+    if (node.ContainsSubgraph()) {
+      const auto subgraphs = node.GetSubgraphs();
+      for (const auto& subgraph : subgraphs) {
+        ORT_RETURN_IF_ERROR(RegisterGraphNodeOpSchemas(*subgraph));
+      }
+    }
+  }
+  return Status::OK();
+}
+
+Status KernelTypeStrResolver::SaveToOrtFormat(
+    fb::FlatBufferBuilder& builder,
+    fb::Offset<fbs::KernelTypeStrResolver>& fbs_kernel_type_str_resolver) const {
+  std::vector<fb::Offset<fbs::OpIdKernelTypeStrArgsEntry>> fbs_op_kernel_type_str_args{};
+  fbs_op_kernel_type_str_args.reserve(op_kernel_type_str_map_.size());
+
+  for (const auto& [op_id, kernel_type_str_map] : op_kernel_type_str_map_) {
+    std::vector<fb::Offset<fbs::KernelTypeStrArgsEntry>> fbs_kernel_type_str_args{};
+    fbs_kernel_type_str_args.reserve(kernel_type_str_map.size());
+
+    for (const auto& [kernel_type_str, args] : kernel_type_str_map) {
+      std::vector<fb::Offset<fbs::ArgTypeAndIndex>> fbs_args{};
+      fbs_args.reserve(args.size());
+
+      for (const auto& arg : args) {
+        auto fbs_arg = fbs::CreateArgTypeAndIndex(
+            builder,
+            arg.first == ArgType::kInput ? fbs::ArgType::INPUT : fbs::ArgType::OUTPUT,
+            gsl::narrow<uint32_t>(arg.second));
+        fbs_args.push_back(fbs_arg);
+      }
+
+      auto fbs_kernel_type_str_args_entry = fbs::CreateKernelTypeStrArgsEntry(
+          builder,
+          builder.CreateSharedString(kernel_type_str),
+          builder.CreateVector(fbs_args));
+      fbs_kernel_type_str_args.push_back(fbs_kernel_type_str_args_entry);
+    }
+
+    fb::Offset<flatbuffers::String> fbs_op_id{};
+    ORT_RETURN_IF_ERROR(fbs::utils::SaveOpIdentifierOrtFormat(builder, op_id, fbs_op_id));
+
+    auto fbs_op_kernel_type_str_args_entry = fbs::CreateOpIdKernelTypeStrArgsEntry(
+        builder,
+        fbs_op_id,
+        builder.CreateVectorOfSortedTables(&fbs_kernel_type_str_args));
+    fbs_op_kernel_type_str_args.push_back(fbs_op_kernel_type_str_args_entry);
+  }
+
+  fbs_kernel_type_str_resolver = fbs::CreateKernelTypeStrResolver(
+      builder,
+      builder.CreateVectorOfSortedTables(&fbs_op_kernel_type_str_args));
+  return Status::OK();
+}
+#endif  // !defined(ORT_MINIMAL_BUILD)
+
+Status KernelTypeStrResolver::LoadFromOrtFormat(const fbs::KernelTypeStrResolver& fbs_kernel_type_str_resolver) {
+  const auto* fbs_op_kernel_type_str_args = fbs_kernel_type_str_resolver.op_kernel_type_str_args();
+  ORT_FORMAT_RETURN_IF_NULL(fbs_op_kernel_type_str_args, "op_kernel_type_str_args");
+
+  OpKernelTypeStrMap op_kernel_type_str_map{};
+  op_kernel_type_str_map.reserve(fbs_op_kernel_type_str_args->size());
+  for (const auto* fbs_op_kernel_type_str_args_entry : *fbs_op_kernel_type_str_args) {
+    ORT_FORMAT_RETURN_IF_NULL(fbs_op_kernel_type_str_args_entry, "op_kernel_type_str_args entry");
```

Comment:
it's a bit tricky with the nested for loops here

Reply:
yeah - that doesn't help. leave it for now but keep in mind it's probably quite expensive binary size wise to have so many conditional exits with different calls to the Status ctor so if we do need to do some reductions this would be one place to spend some time on.